resolwe_bio/processes/import_data/metadata.py

```diff
@@ -1,93 +1,161 @@
-        data.has_missing_class() and data.get_nan_frequency_attribute()
-    )
+    slug = "upload-metadata-unique"
+    name = "Metadata table (one-to-one)"
+    process_type = "data:metadata:unique"
+    version = "1.0.0"
+    category = "Import"
+    scheduling_class = SchedulingClass.BATCH
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/s4q6j6e8/resolwebio/orange:2.0.0"}
+        },
+        "resources": {"cores": 1, "memory": 8192},
+    }
+    data_name = '{{ src.file|default("?") }}'
 
-    if data.domain.has_continuous_class:
-        target_description = f"Regression; numerical class {missing_in_class}"
-    elif data.domain.has_discrete_class:
-        target_description = (
-            "Classification; categorical class "
-            f"with {len(data.domain.class_var.values)} values {missing_in_class}"
-        )
-    elif data.domain.class_vars:
-        target_description = (
-            "Multi-target; "
-            f"{len(data.domain.class_vars)} target variables "
-            f"{missing_in_class}"
+    class Input:
+        """Input field to process UploadMetadataUnique."""
+
+        src = FileField(
+            label="Table with metadata",
+            description="The table should be in Orange format and use "
+            "one of the following extensions: .csv, .tab, .tsv, .xlsx, "
+            ".xls",
         )
-    else:
-        target_description = "Not defined"
 
-    return target_description
+    class Output:
+        """Output field of the process UploadMetadataUnique."""
 
+        table = FileField(label="Uploaded table")
+        n_samples = IntegerField(label="Number of samples")
 
-def change_suffix(path):
-    """Change suffix of a file to lowercase."""
-    new_path = path.with_suffix(path.suffix.lower())
-    path.replace(new_path)
-    return new_path
+    def run(self, inputs, outputs):
+        """Run the analysis."""
 
+        collections = Collection.filter(data__id=self.data.id)
+        if not collections:
+            self.error(
+                "Metadata table was not uploaded to a Collection. "
+                "Matching of metadata entries to Sample objects is not possible."
+            )
 
-class UploadOrangeMetadata(Process):
-    """Upload metadata table in Orange format.
+        path = Path(inputs.src.import_file(imported_format="extracted"))
 
-    Orange can read files in native tab-delimited format, or can load
-    data from any of the major standard spreadsheet file types, like CSV
-    and Excel. Native format starts with the names of attributes with
-    prefixes that define attribute type (continuous, discrete, time,
-    string) and role (class, meta, ignore, instance weights). Prefixes
-    are separated from the attribute name with a hash sign (“#”).
+        # change the file suffix if it is either upper or mixed case
+        if not path.suffix.islower():
```

Comment:
You could pack all the logic here in function `change_suffix` (I would rather name it `lower_suffix`, more descriptive). This function would just return a path if it was lowercase, but otherwise do all the path fixing and info reporting.

Since the logic is duplicated in both processes this would also shorten the code.